_inc/lib/core-api/class.jetpack-core-api-module-endpoints.php

```diff
@@ -1066,6 +1075,95 @@ private function _process_onboarding( $data ) {
 			: join( ', ', $error );
 	}
 
+	/**
+	 * Retrieve the first active sidebar.
+	 *
+	 * @return array|WP_Error First active sidebar, error if none exists.
+	*/
+	static function get_first_sidebar() {
+		$active_sidebars = get_option( 'sidebars_widgets', array() );
+		unset( $active_sidebars[ 'wp_inactive_widgets' ], $active_sidebars[ 'array_version' ] );
+
+		if ( empty( $active_sidebars ) ) {
+			return new WP_Error( 'invalid_data', 'Invalid data.', 400 );
+		}
+		$active_sidebars_keys = array_keys( $active_sidebars );
+		return array_shift( $active_sidebars_keys );
+	}
+
+	/**
+	 * Add or update Business Address widget.
+	 *
+	 * @param array $business_address_content Value corresponding to the Business
+	 * Address as received from the settings endpoint.
+	 *
+	 * @return WP_Error|true True when data has been saved correctly,
+	 * error otherwise.
+	*/
+	static function handle_business_address( $business_address_content ) {
+		$first_sidebar = self::get_first_sidebar();
+
+		$widgets_module_active = Jetpack::is_module_active( 'widgets' );
+		if ( ! $widgets_module_active ) {
+			$widgets_module_active = Jetpack::activate_module( 'widgets', false, false );
+		}
+		if ( ! $widgets_module_active ) {
+			return new WP_Error( 'invalid_data', 'Failed to activate module.', 400 );
+		}
+
+		if ( $first_sidebar ) {
+			$title = $business_address_content['name'];
+			$address =
+				$business_address_content['city'] . ' ' .
+				$business_address_content['state'] . ' ' .
+				$business_address_content['street'] . ' ' .
+				$business_address_content['zip'];
+			$widget_options = array(
+				'title'   => $title,
+				'address' => $address,
+				'phone'   => '',
+				'hours'   => '',
+				'showmap' => false,
+				'email' => ''
+			);
+
+			$widget_inserted = '';
+			$widget_updated = '';
+			if ( ! self::has_business_address_widget( $first_sidebar ) ) {
+				$widget_inserted = Jetpack_Widgets::insert_widget_in_sidebar( 'widget_contact_info', $widget_options, $first_sidebar );
+			} else {
+				$widget_updated = Jetpack_Widgets::update_widget_in_sidebar( 'widget_contact_info', $widget_options, $first_sidebar );
+			}
+			if ( is_wp_error( $widget_inserted ) || is_wp_error( $widget_updated ) ) {
+				return new WP_Error( 'invalid_data', 'Invalid update.', 400 );
+			}
+
+			return true;
+		}
+		// No sidebar to place the widget
+		return new WP_Error( 'invalid_data', 'Invalid data.', 400 );
+	}
+
+	/**
+	 * Check if a given sidebar already contains a widget with a Business Address.
```

Comment:
We don't really check whether we have a Business Address; rather we check whether there's a Jetpack "Contact Info & Map" (`widget_contact_info`) widget in the given sidebar.